src/middleware.ts

```diff
@@ -0,0 +1,38 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { getToken } from 'next-auth/jwt';
+import { Role } from './constants/Enum/role.enum';
+
+export async function middleware(request: NextRequest) {
+  const redirectUrl = request.nextUrl;
+  const hasSessionToken = request.cookies.has('next-auth.session-token');
+  const isSignInRoute = redirectUrl.pathname === '/auth/sign-in';
+  const protectedPaths = ['/admin'];
+  const matchesProtectedPath = protectedPaths.some((path) =>
+    redirectUrl.pathname.startsWith(path),
+  );
+  const token = await getToken({ req: request });
+  if (hasSessionToken) {
```

Comment:
const redirectUrl = new URL(request.url);
const isAdmin = hasSessionToken && matchesProtectedPath && token.zone[0] === Role.ADMIN;

if (isAdmin) {
  return NextResponse.next();
}

if (!hasSessionToken) {
  redirectUrl.pathname = '/auth/sign-in';
} else {
  redirectUrl.pathname = '/landing-page';
}

return NextResponse.redirect(redirectUrl);